config/file/parse.go

```diff
@@ -42,41 +44,116 @@ type unmarshaledConfig struct {
 	} `yaml:"output" json:"output"`
 }
 
-// Parse parses a benchttp runner config file into a config.Config
-// and returns it or the first non-nil error occurring in the process.
+// Parse parses a benchttp runner config file into a config.Global
+// and returns it or the first non-nil error occurring in the process,
+// which can be any of the values declared in the package.
 func Parse(cfgpath string) (cfg config.Global, err error) {
+	uconfs, err := parseFileRecursive(cfgpath, []unmarshaledConfig{}, map[string]bool{})
+	if err != nil {
+		return cfg, err
+	}
+	return parseAndMergeConfigs(uconfs)
+}
+
+// parseFileRecursive parses a config file and its parent found from key
+// "extends" recursively until the root config file is reached.
+// It returns the list of all parsed configs or the first non-nil error
+// occurring in the process.
+func parseFileRecursive(
+	cfgpath string,
+	uconfs []unmarshaledConfig,
+	seen map[string]bool,
+) ([]unmarshaledConfig, error) {
+	// parse current config file
+	uconf, err := parseFile(cfgpath)
+	if err != nil {
+		return uconfs, err
+	}
+
+	// root config reached, append it and return the result
+	if uconf.Extends == nil {
+		uconfs = append(uconfs, uconf)
+		return uconfs, nil
+	}
+
+	// avoid infinite recursion caused by circular reference
+	if _, exists := seen[*uconf.Extends]; exists {
+		return uconfs, ErrCircularExtends
+	}
+
+	// record file, append config
+	seen[*uconf.Extends] = true
+	uconfs = append(uconfs, uconf)
+
+	// resolve extended config path
+	parentPath := path.Join(path.Dir(cfgpath), *uconf.Extends)
```

Comment:
What does it do if the the paths are mixed between relative and absolute ?

Not that we should handle that case, but at least return an error to the user.

Reply:
Nested paths are correctly resolved, absolute path are not handled (it would require us to be opinionated about what is the root).

In such cases, a relevant `ErrNotExists` is returned by `parseFile`